distributed/tests/test_config.py

```diff
@@ -338,13 +338,14 @@ def test_schema_is_complete():
         schema = yaml.safe_load(f)
 
     skip = {
-        "default-task-durations",
-        "bokeh-application",
-        "environ",
-        "pre-spawn-environ",
+        "distributed.scheduler.default-task-durations",
```

Comment:
Why are these changes needed?

Reply:
The schema validation in this test traverses the full yaml schema from its root to any leaves. Normally, any `object` property (mapping to a `dict` in python) is required to have `properties` set (indicating which keys are valid in the dict). In some cases (for example the nanny environment) the keys can be arbitrary, so we skip validation. 

Previously this skipping was just done on the leaf key name (disregarding the path to the leaf), so could have resulted in false positive skipping (e.g. previously _any_ key name `environ` would not have had its properties validated even in the case where they existed).

To avoid this, I'm changing the skip list to be specify the fully-qualified key names, and remembering the path to the leaf when traversing.